### PR TITLE
Codepipeline

### PR DESCRIPTION
**Pull request by abrenaut**
Use CodePipeline to kick off CodeBuild custom raspbian image creation process
Use latest version of aws-iot-fleet-provisioning reference client
Use least privilege policy for IoT things

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/aws-iot-rpi-fleet-provisioning-stack.ts [hunk 1]
```diff
@@ -146,42 +179,76 @@ export class AwsIotRpiFleetProvisioningStack extends cdk.Stack {
...
+      timeout: cdk.Duration.seconds(60),
+      initialPolicy: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'iot:CreateKeysAndCertificate',
+            'iot:AttachPolicy',
+            'iot:DescribeEndpoint',
+          ],
+          resources: ['*'],
```
Comment (martenpayne) [inline lib/aws-iot-rpi-fleet-provisioning-stack.ts:209]: un-scoped resource access?

File: lib/lambda/configure_rpi_image_builder/app.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ROOT_CERT = root.ca.pem
+CLAIM_CERT = {CERT_FILENAME}
+SECURE_KEY = {KEY_FILENAME}
+
+# Set the name of your IoT Endpoint
+IOT_ENDPOINT = {get_iot_endpoint()}
+
+# Include the name for the provisioning template that was created in IoT Core
+PRODUCTION_TEMPLATE = {provisioning_template_name}
+# TODO: Implement cert rotation
```
Comment (martenpayne) [inline lib/lambda/configure_rpi_image_builder/app.py]: careful of leaving TODOs in public code

File: lib/lambda/configure_rpi_image_builder/app.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    configured_rpi_image_builder_object_key,
+):
+
+    logger.info("Downloading client code")
+    rpi_image_builder_object = s3_client.get_object(
+        Bucket=rpi_image_builder_bucket_name,
+        Key=rpi_image_builder_object_key,
+    )
+
+    with io.BytesIO(rpi_image_builder_object['Body'].read()) as tf:
```
Comment (martenpayne) [inline lib/lambda/configure_rpi_image_builder/app.py]: tf is an obtuse var name

File: lib/lambda/configure_rpi_image_builder/app.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    create_configured_rpi_image_builder_archive(
+        claim_certificate,
+        config,
+        rpi_image_builder_bucket_name,
+        rpi_image_builder_object_key,
+        configured_rpi_image_builder_bucket_name,
+        configured_rpi_image_builder_object_key,
+    )
+
+
```
Comment (martenpayne) [inline lib/lambda/configure_rpi_image_builder/app.py:125]: on what event? … maybe a comment for this method

File: rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
+# ------------------------------------------------------------------------------
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0 
+# ------------------------------------------------------------------------------
+# Demonstrates how to call/orchestrate AWS fleet provisioning services
+#  with a provided bootstrap certificate (aka - provisioning claim cert).
+#   
+# Initial version - Raleigh Murch, AWS
```
Comment (martenpayne) [inline rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py]: do we need to include these personal contact detail?

File: rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py [hunk 6]
```diff
@@ -0,0 +1,339 @@
+# ------------------------------------------------------------------------------
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
```
Comment (martenpayne) [inline rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py:3]: this is MIT-0, while the rest of the source is Apache 2.0…

**Review by martenpayne — COMMENTED** [review]
A few comments - nothing too major

File: lib/aws-iot-rpi-fleet-provisioning-stack.ts (hunk 1 above)
Comment (abrenaut) [reply lib/aws-iot-rpi-fleet-provisioning-stack.ts:209]: I don't see how the scope of those three actions could be restricted in this context

File: rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py (hunk 6 above)
Comment (abrenaut) [reply rpi-image-builder/aws-iot-fleet-provisioning/provisioning_handler.py:3]: This code comes from a different project That's using MIT-0 license

File: lib/aws-iot-rpi-fleet-provisioning-stack.ts (hunk 1 above)
Comment (martenpayne) [reply lib/aws-iot-rpi-fleet-provisioning-stack.ts:209]: yes, now I see the context - you're right

**Comment by martenpayne** [comment]
looks good to me 👍

**Comment by martenpayne** [comment]
looks good to me 👍

**Review by martenpayne — APPROVED** [review]
good to go

**Pull request merged**